### PR TITLE
[4.0] Cleanup privacy menu

### PR DESCRIPTION
**Pull request by roland-d**
### Summary of Changes
Moves the submenu into the main left sidebar:
image

The layouts are updated using the namespaces.

### Testing Instructions
1. Login to the backend
2. Click on Privacy under Users
3. Notice the dashboard opens with the submenu in it.
4. Click the submenu items, the same submenu is visible on these pages
5. Apply pull request
6. Click Privacy under Users
7. A submenu folds open with the Privacy views
8. Click each view to see if it works

### Expected result
Menu items below Privacy

### Actual result
Menu items on the view page

### Documentation Changes Required
None

@wilsonge

### CODE REVIEW

File: administrator/components/com_menus/presets/joomla.xml [hunk 1]
```diff
@@ -1,255 +1,281 @@
...
-		xmlns="urn:joomla.org"
-		xsi:schemaLocation="urn:joomla.org menu.xsd">
-	<menuitem
-			type="component"
-			title="MOD_MENU_CONTROL_PANEL"
-			link="index.php"
-			class="class:home"
-			element="com_cpanel"
-	/>
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (wilsonge) [inline administrator/components/com_menus/presets/joomla.xml]: Back to tabs please :) sorry!

File: administrator/components/com_privacy/View/Request/HtmlView.php [hunk 2]
```diff
@@ -64,17 +65,18 @@ class HtmlView extends BaseHtmlView
 	 *
 	 * @param   string  $tpl  The name of the template file to parse; automatically searches through the template paths.
 	 *
-	 * @return  mixed  A string if successful, otherwise an Error object.
+	 * @return  void
 	 *
+	 * @throws  \Exception
 	 * @see     BaseHtmlView::loadTemplate()
 	 * @since   3.9.0
-	 * @throws  Exception
```
Comment (mbabker) [inline administrator/components/com_privacy/View/Request/HtmlView.php:71]: A code style rule really needs to be enforced here because people randomly put tags in whatever order, including randomly rearranging them, and those kinds of changes are just pointless.  So either there is a PHPCS rule enforcing `@throws` before `@since` or this change should be reverted until a rule is created.

File: administrator/components/com_privacy/View/Request/HtmlView.php [hunk 3]
```diff
@@ -64,17 +65,18 @@ class HtmlView extends BaseHtmlView
...
 	 */
 	public function display($tpl = null)
 	{
-		// Initialise variables.
-		$this->item  = $this->get('Item');
-		$this->state = $this->get('State');
+		/** @var RequestsModel $model */
+		$model       = $this->getModel();
+		$this->item  = $model->getItem();
```
Comment (mbabker) [inline administrator/components/com_privacy/View/Request/HtmlView.php:78]: All of these changes about how model data is pulled seems unrelated to this pull request, and as far as I'm aware use of `AbstractView::get()` is not deprecated.

File: administrator/components/com_privacy/tmpl/consents/default.php [hunk 4]
```diff
@@ -7,65 +7,68 @@
...
+HTMLHelper::_('formbehavior.chosen', 'select');
 
 $user       = JFactory::getUser();
 $listOrder  = $this->escape($this->state->get('list.ordering'));
 $listDirn   = $this->escape($this->state->get('list.direction'));
 $now        = JFactory::getDate();
 $stateIcons = array(-1 => 'trash', 0 => 'archive', 1 => 'publish');
-$stateMsgs  = array(-1 => JText::_('COM_PRIVACY_CONSENTS_STATE_INVALIDATED'), 0 => JText::_('COM_PRIVACY_CONSENTS_STATE_OBSOLETE'), 1 => JText::_('COM_PRIVACY_CONSENTS_STATE_VALID'));
+$stateMsgs  = array(
+		-1 => Text::_('COM_PRIVACY_CONSENTS_STATE_INVALIDATED'),
```
Comment (mbabker) [inline administrator/components/com_privacy/tmpl/consents/default.php]: Arrays should be indented one level, not two.

File: administrator/components/com_privacy/View/Request/HtmlView.php (hunk 2 above)
Comment (roland-d) [reply administrator/components/com_privacy/View/Request/HtmlView.php:71]: Changed it in all views after the since tag

File: administrator/components/com_privacy/View/Request/HtmlView.php (hunk 3 above)
Comment (roland-d) [reply administrator/components/com_privacy/View/Request/HtmlView.php:78]: Well the get method makes it impossible for at least my IDE to figure out what we are trying to do. You are not using the get method either in the XmlView. I prefer the IDE understanding our code over not knowing what is going on.

File: administrator/components/com_privacy/tmpl/consents/default.php (hunk 4 above)
Comment (roland-d) [reply administrator/components/com_privacy/tmpl/consents/default.php]: Done.

File: administrator/components/com_privacy/View/Request/HtmlView.php (hunk 3 above)
Comment (mbabker) [reply administrator/components/com_privacy/View/Request/HtmlView.php:78]: I prefer it too.  I'd go so far as to say deprecate `AbstractView::get()` and force this convention for exactly the reasons you just said.  But, it's not the convention used in core right now and we already have enough problems of different components doing things differently because every contributor has a different coding style and preference.

FWIW the use in the export's XML view is because `AbstractView::get()` only works for methods prefixed with get, i.e. `getItem`.  It won't work with `collectDataForExportRequest()` as a result and I didn't feel like naming the method a certain way just to work with another unrelated API's limitation.

Comment (roland-d) [reply administrator/components/com_privacy/View/Request/HtmlView.php:78]: It may not be the convention now but I don't mind cleaning it up across the board. Once we have that we can say that it is convention. I spoke with @wilsonge yesterday that the docblocks have no codestyle but I would love to see us set a fixed format because now nobody knows what to do.

Comment (mbabker) [reply administrator/components/com_privacy/View/Request/HtmlView.php:78]: Whether we have consistency on crap conventions or consistency on helpful conventions, I'm happy with consistent code 😉 (and after years of writing code working with it, I think the fact it's perceived you have to prefix model methods with `get` because of the "magic" of the view's `get` method is a bad convention that needs to die, but I've always favored a more explicit code convention that relies less on magic behaviors).

**Comment by roland-d** [comment]
@wilsonge Anything else needed here?

File: administrator/components/com_privacy/View/Dashboard/HtmlView.php [hunk 5]
```diff
@@ -64,6 +65,14 @@ class HtmlView extends BaseHtmlView
 	 */
 	protected $sendMailEnabled;
 
+	/**
+	 * Days when a request is considered urgent
+	 *
+	 * @var    integer
+	 * @since  3.9.0
+	 */
+	protected $urgentRequestDays = 14;
```
Comment (wilsonge) [inline administrator/components/com_privacy/View/Dashboard/HtmlView.php:77]: Why are we introducing a variable here (it's definitely not 3.9.0)?

Comment (roland-d) [reply administrator/components/com_privacy/View/Dashboard/HtmlView.php:77]: @wilsonge Not introducing a variable here. This variable was undefined on line 115:
https://github.com/joomla/joomla-cms/pull/24810/files/a1a97099f8c89892ef9e0d1395d7b6dcb10f8d45#diff-4cbbd6bc0abb339b293d4ddd9df2cbddR115

I merely documented it with a docblock.

File: administrator/components/com_privacy/tmpl/consents/default.php [hunk 6]
```diff
@@ -7,65 +7,68 @@
  * @license     GNU General Public License version 2 or later; see LICENSE.txt
  */
 
+use Joomla\CMS\Date\Date;
```
Comment (Quy) [inline administrator/components/com_privacy/tmpl/consents/default.php:12]: Move `use` statements after `defined`. Repeat in other files.

Comment (roland-d) [reply administrator/components/com_privacy/tmpl/consents/default.php:12]: Done.

File: administrator/components/com_privacy/tmpl/request/edit.php [hunk 7]
```diff
@@ -7,13 +7,17 @@
  * @license     GNU General Public License version 2 or later; see LICENSE.txt
  */
 
+use Joomla\CMS\Factory;
+use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Router\Route;
+
 defined('_JEXEC') or die;
```
Comment (Quy) [inline administrator/components/com_privacy/tmpl/request/edit.php]: Move above `use` statements.

File: administrator/language/en-GB/en-GB.mod_menu.ini [hunk 8]
```diff
@@ -25,6 +25,10 @@ MOD_MENU_COM_LANGUAGES_SUBMENU_CONTENT="Content Languages"
 MOD_MENU_COM_LANGUAGES_SUBMENU_INSTALLED="Installed"
 MOD_MENU_COM_LANGUAGES_SUBMENU_OVERRIDES="Overrides"
 MOD_MENU_COM_PRIVACY="Privacy"
+MOD_MENU_COM_PRIVACY_DASHBOARD="Dashboard"
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.mod_menu.ini]: Sort key in alpha order.

Comment (roland-d) [reply administrator/language/en-GB/en-GB.mod_menu.ini]: Done

File: administrator/components/com_privacy/tmpl/request/edit.php (hunk 7 above)
Comment (roland-d) [reply administrator/components/com_privacy/tmpl/request/edit.php]: Done

File: administrator/components/com_privacy/tmpl/requests/default.php [hunk 9]
```diff
@@ -9,60 +9,61 @@
...
-		<div id="j-sidebar-container" class="span2">
-			<?php echo $this->sidebar; ?>
-		</div>
-		<div id="j-main-container" class="span10">
-	<?php else : ?>
-		<div id="j-main-container">
-	<?php endif; ?>
-		<?php echo JLayoutHelper::render('joomla.searchtools.default', array('view' => $this)); ?>
+<form action="<?php echo Route::_('index.php?option=com_privacy&view=requests'); ?>" method="post" name="adminForm" id="adminForm">
+    <div id="j-main-container">
```
Comment (Quy) [inline administrator/components/com_privacy/tmpl/requests/default.php]:
```suggestion
	<div id="j-main-container">
```

File: administrator/components/com_privacy/View/Capabilities/HtmlView.php [hunk 10]
```diff
@@ -74,7 +74,7 @@ public function display($tpl = null)
 
 		$this->sidebar = \JHtmlSidebar::render();
```
Comment (Quy) [inline administrator/components/com_privacy/View/Capabilities/HtmlView.php]: Delete

Comment (Quy) [reply administrator/components/com_privacy/View/Capabilities/HtmlView.php]: Delete 

https://github.com/joomla/joomla-cms/blob/b7059667ec38aa6e7c045ec60da72d1efbee8bed/administrator/components/com_privacy/View/Capabilities/HtmlView.php#L34-L40

File: administrator/components/com_privacy/View/Consents/HtmlView.php [hunk 11]
```diff
@@ -104,9 +106,7 @@ public function display($tpl = null)
 
 		$this->addToolbar();
 
-		$this->sidebar = \JHtmlSidebar::render();
```
Comment (Quy) [inline administrator/components/com_privacy/View/Consents/HtmlView.php:107]: Delete 
https://github.com/joomla/joomla-cms/blob/b7059667ec38aa6e7c045ec60da72d1efbee8bed/administrator/components/com_privacy/View/Consents/HtmlView.php#L64-L70

File: administrator/components/com_privacy/View/Dashboard/HtmlView.php [hunk 12]
```diff
@@ -118,9 +128,7 @@ public function display($tpl = null)
 
 		$this->addToolbar();
 
-		$this->sidebar = \JHtmlSidebar::render();
```
Comment (Quy) [inline administrator/components/com_privacy/View/Dashboard/HtmlView.php:121]: Delete
https://github.com/joomla/joomla-cms/blob/b7059667ec38aa6e7c045ec60da72d1efbee8bed/administrator/components/com_privacy/View/Dashboard/HtmlView.php#L79-L85

File: administrator/components/com_privacy/View/Requests/HtmlView.php [hunk 13]
```diff
@@ -123,9 +125,7 @@ public function display($tpl = null)
 
 		$this->addToolbar();
 
-		$this->sidebar = \JHtmlSidebar::render();
```
Comment (Quy) [inline administrator/components/com_privacy/View/Requests/HtmlView.php:126]: Delete 
https://github.com/joomla/joomla-cms/blob/b7059667ec38aa6e7c045ec60da72d1efbee8bed/administrator/components/com_privacy/View/Requests/HtmlView.php#L73-L79

**Comment by Quy** [comment]
2 menus should be under `Users`.
users-menu

**Comment by wilsonge** [comment]
@roland-d if you fix the bugs @Quy mentions above and fix the conflicts let's get this in

**Comment by roland-d** [comment]
@wilsonge All changes are done. Let's see if nothing else is found ;)

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on f1eba227f2ff24f788043344f1e2a7e5ccbcfe65<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24810">issues.joomla.org/tracker/joomla-cms/24810</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Comment by roland-d** [comment]
Thank you.

**Comment by brianteeman** [comment]
Thanks
